pkg/snmp/snmp.go

```diff
@@ -144,20 +144,40 @@ func (c *Config) BuildSNMPParams() (*gosnmp.GoSNMP, error) {
 
 	var authProtocol gosnmp.SnmpV3AuthProtocol
 	lowerAuthProtocol := strings.ToLower(c.AuthProtocol)
-	if lowerAuthProtocol == "md5" {
+	if lowerAuthProtocol == "none" {
+		authProtocol = gosnmp.NoAuth
+	} else if lowerAuthProtocol == "md5" {
 		authProtocol = gosnmp.MD5
 	} else if lowerAuthProtocol == "sha" {
 		authProtocol = gosnmp.SHA
+	} else if lowerAuthProtocol == "sha224" {
+		authProtocol = gosnmp.SHA224
+	} else if lowerAuthProtocol == "sha256" {
+		authProtocol = gosnmp.SHA256
+	} else if lowerAuthProtocol == "sha384" {
+		authProtocol = gosnmp.SHA384
+	} else if lowerAuthProtocol == "sha512" {
+		authProtocol = gosnmp.SHA512
 	} else if c.AuthProtocol != "" {
 		return nil, fmt.Errorf("Unsupported authentication protocol: %s", c.AuthProtocol)
 	}
```

Comment:
Continuing on that comment, `gosnmp.NoAuth` equals 1: https://godoc.org/github.com/soniah/gosnmp#SnmpV3AuthProtocol

Further down we set `msgFlags = gosnmp.AuthNoPriv` https://github.com/DataDog/datadog-agent/pull/5632/files#diff-e3d79654e541e8cdc1941ea637ff3de3R188 on an non-empty lowerAuthProtocol
gosnmp will raise an error if a user sets `authentication_protocol: none` in their config https://github.com/soniah/gosnmp/blob/master/v3_usm.go#L267

I think the correct approach would be something like this:
```suggestion
	if lowerAuthProtocol == "" {
		authProtocol = gosnmp.NoAuth
	} else if lowerAuthProtocol == "md5" {
		authProtocol = gosnmp.MD5
	} else if lowerAuthProtocol == "sha" {
		authProtocol = gosnmp.SHA
	} else if lowerAuthProtocol == "sha224" {
		authProtocol = gosnmp.SHA224
	} else if lowerAuthProtocol == "sha256" {
		authProtocol = gosnmp.SHA256
	} else if lowerAuthProtocol == "sha384" {
		authProtocol = gosnmp.SHA384
	} else if lowerAuthProtocol == "sha512" {
		authProtocol = gosnmp.SHA512
	} else {
		return nil, fmt.Errorf("Unsupported authentication protocol: %s", c.AuthProtocol)
	}
```

Same comment for privacy protocol